packages/dapp/src/components/HighlightModal/Footer.js

```diff
@@ -19,23 +19,35 @@ const Footer = () => {
       // throw an error here or prompt the user to reload
       return;
     }
+    const image = state.image.split(",")[1]
+    const { arUrl: imageUrl } = (await upload({
+        data: image,
+        contentType: 'image/png'
+      })).data;
 
-    // fetch and upload image
-    const image = Buffer.from(state.image.split(",")[1], "base64")
-    const { arUrl: imageUrl } = await upload({ data: image, contentType: 'image/png' });
+    const hash = keccak256(defaultAbiCoder.encode(['string'],[state.text]));
 
-    // fetch, generate and upload metadata
-    const tokenId = nextTokenId(provider);
-    const metadata = generate(tokenId, imageUrl, state.text.length);
-    const { arUrl: metadataUrl } = await upload({
-      data: JSON.stringify(metadata),
-      contentType: 'text/plain'
+    const metadata = generate(hash, imageUrl, state.text.length);
+
+    const { arUrl: metadataUrl } = (await upload({
+        data: JSON.stringify(metadata),
+        contentType: 'text/plain'
+      })).data;
+
+    const { signature, id } = (await sign({ arUrl: metadataUrl })).data;
+
+    await mintSelected({
+      url: metadataUrl,
+      provider,
+      signer,
+      id,
+      signature
     });
 
-    console.log(metadataUrl);
+    const owner = await ownerOf(provider, id);
```

Comment:
why do we need this?

I especially don't like the name (or the name of the route in the OZ contracts, given the content of the essay)

Reply:
this is just for now, we wouldn't need this in the final version, once we have decided what to display right after the mint